src/reducers/rooms.reducer.js

```diff
@@ -1,196 +1,44 @@
-        start: new Date(2018, 1, 23, 13, 0, 0),
-        end: new Date(2018, 1, 23, 14, 0, 0)
-      },
-      {
-        id: 2,
-        title: "DTS STARTS",
-        start: new Date(2018, 2, 13, 10, 0, 0),
-        end: new Date(2018, 2, 13, 11, 0, 0)
-      },
-    
-      {
-        id: 3,
-        title: "DTS ENDS",
-        start: new Date(2018, 10, 6, 0, 0, 0),
-        end: new Date(2018, 10, 13, 0, 0, 0)
-      },
-    
-      {
-        id: 4,
-        title: "Some Event",
-        start: new Date(2018, 3, 9, 0, 0, 0),
-        end: new Date(2018, 3, 9, 0, 0, 0)
-      },
-      {
-        id: 5,
-        title: "Conference",
-        start: new Date(2018, 3, 11),
-        end: new Date(2018, 3, 13),
-        desc: "Big conference for important people"
-      },
-      {
-        id: 6,
-        title: "Meeting",
-        start: new Date(2018, 3, 12, 10, 30, 0, 0),
-        end: new Date(2018, 3, 12, 12, 30, 0, 0),
-        desc: "Pre-meeting meeting, to prepare for the meeting"
-      },
-      {
-        id: 7,
-        title: "Lunch",
-        start: new Date(2018, 3, 12, 12, 0, 0, 0),
-        end: new Date(2018, 3, 12, 13, 0, 0, 0),
-        desc: "Power lunch"
-      },
-      {
-        id: 8,
-        title: "Meeting",
-        start: new Date(2018, 3, 12, 14, 0, 0, 0),
-        end: new Date(2018, 3, 12, 15, 0, 0, 0)
-      },
-      {
-        id: 9,
-        title: "Happy Hour",
-        start: new Date(2018, 3, 12, 17, 0, 0, 0),
-        end: new Date(2018, 3, 12, 17, 30, 0, 0),
-        desc: "Most important meal of the day"
-      },
-      {
-        id: 10,
-        title: "Dinner",
-        start: new Date(2018, 3, 12, 20, 0, 0, 0),
-        end: new Date(2018, 3, 12, 21, 0, 0, 0)
-      },
-      {
-        id: 11,
-        title: "Birthday Party",
-        start: new Date(2019, 3, 13, 7, 0, 0),
-        end: new Date(2018, 3, 13, 10, 30, 0)
-      },
-      {
-        id: 12,
-        title: "Late Night Event",
-        start: new Date(2018, 3, 17, 19, 30, 0),
-        end: new Date(2018, 3, 18, 2, 0, 0)
-      },
-      {
-        id: 13,
-        title: "Multi-day Event",
-        start: new Date(2018, 3, 20, 19, 30, 0),
-        end: new Date(2018, 3, 22, 2, 0, 0)
-      }],
-      isActive: false,
-      backgroundImage: 'ayleysbury.jpg',
-      barImage: 'bar2.png',
-    },
-    {
-      id: 'haywards',
-      label: 'Hayward’s (née MR2) Room',
-      appointments: [],
-      isActive: false,
-      backgroundImage: 'haywards.jpg',
-      barImage: 'bar.png',
-    },
-    {
-      id: 'sheep',
-      label: 'Sheep',
-      appointments: [],
-      isActive: false,
-      backgroundImage: 'sheep.jpg',
-      barImage: 'bar2.png',
-    },
-    {
-      id: 'kitchen',
-      label: 'Kitchen',
-      appointments: [],
-      isActive: false,
-      backgroundImage: 'kitchen.jpg',
-      barImage: 'bar.png',
-    },
-  ]
-}
+const initialState = { rooms: roomData };
 
```

Comment:
Probably the biggest improvement in my opinion.

Previously the action.type would be mapped through an array of reducers until it hit the correct one.
Now, our reducers are keys in an object. Benefits of this are no mapping, just a key to select the reducer. Much faster and less code.

[https://redux-actions.js.org/api-reference/handleaction-s]